clippy_lints/src/attrs/duplicated_attributes.rs

```diff
@@ -35,30 +36,38 @@ fn check_duplicated_attr(
     if attr.span.from_expansion() {
         return;
     }
-    let Some(ident) = attr.ident() else { return };
-    let name = ident.name;
-    if name == sym::doc || name == sym::cfg_attr_trace || name == sym::rustc_on_unimplemented || name == sym::reason {
-        // FIXME: Would be nice to handle `cfg_attr` as well. Only problem is to check that cfg
-        // conditions are the same.
-        // `#[rustc_on_unimplemented]` contains duplicated subattributes, that's expected.
-        return;
-    }
-    if let Some(direct_parent) = parent.last()
-        && *direct_parent == sym::cfg_trace
-        && [sym::all, sym::not, sym::any].contains(&name)
-    {
-        // FIXME: We don't correctly check `cfg`s for now, so if it's more complex than just a one
-        // level `cfg`, we leave.
-        return;
+    let attr_path = path_to_string(&attr.path);
```

Comment:
Given that most attribute paths are probably simple idents, it might make sense to special case idents here rather than on line 66-70, to avoid pretty-printing/allocating in those common cases

```suggestion
    let attr_path = if let Some(ident) = attr.ident() {
        ident.name
    } else {
        Symbol::intern(&path_to_string(&attr.path))
    };
```
(then L66-70 could be removed).

What do you think?